cdap-notifications/src/main/java/co/cask/cdap/notifications/service/inmemory/InMemoryNotificationService.java

```diff
@@ -64,6 +65,10 @@ protected void shutDown() throws Exception {
   public <N> ListenableFuture<N> publish(final Id.NotificationFeed feed, final N notification,
                                          final Type notificationType)
     throws NotificationException {
+    if (executorService == null) {
```

Comment:
In standalone, stream service is started before the app fabric starts and it tries to publish even before the `InMemoryNotificationService.startUp` gets called, which is why the `executorService` is null here. Once the `InMemoryNotificationService.startUp` happens the `executorService` gets initialized and next calls to publish succeeds. This is why I thought of throwing `ServiceUnavailable` here. 
However `IllegalStateException` also makes sense here. Let me know if need to change.